panel/src/components/Forms/Blocks/Blocks.vue

```diff
@@ -358,6 +495,62 @@ export default {
         this.$refs["block-" + block.id][0].open();
       }
     },
+    async paste(e) {
+      let html = e;
+
+      // get the content from the clipboard if an event is given
+      if (e instanceof ClipboardEvent) {
```

Comment:
same here, I would move this to a clipboard paste helper, allows us later to extend the clipboard js functionality without having to jump into the component's code